packages/terra-tabs/src/Tabs.jsx

```diff
@@ -130,16 +154,39 @@ class Tabs extends React.Component {
           showIcon={child.props.showIcon}
           render={() => tabContent}
           isDisabled={child.props.isDisabled}
+          isClosable={isClosable}
           variant="framework"
         />,
       );
     });
+    const handleTabsStateChange = (newValue, itemKey, event) => {
+      if (newValue.length > 0) {
+        let activeAfterClosed = '';
+        for (let i = 0; i < newValue.length; i += 1) {
+          if (newValue[i].isSelected === true) {
+            activeAfterClosed = newValue[i].itemKey;
+            break;
+          }
+        }
+        this.setState({ activeAfterClosed });
+      } else if (newValue.length === 0) {
+        this.setState({ activeAfterClosed: '' });
+      }
+
+      if (this.state.activeKey === itemKey) {
+        setTimeout(() => {
+          this.setState((prevState) => ({ activeKey: prevState.activeAfterClosed }));
+        }, 50);
```

Comment:
instead of time out can we move this change to callback and invoke it synchronous to state update `this.setState({ activeAfterClosed }, callback);`
```suggestion
     const setActiveKey = () => {
      if (this.state.activeKey === itemKey) {
          this.setState((prevState) => ({ activeKey: prevState.activeAfterClosed }));
       }
  };
```
`this.setState({ activeAfterClosed }, setActiveKey);`

Reply:
Thanks done https://github.com/cerner/terra-framework/commit/628052153b88461ef88733be6b05c1fb519e3f41